src/NzbDrone.Core/Extras/Files/AlbumExtraFileManager.cs

```diff
@@ -0,0 +1,164 @@
+                    result.Add(file);
+                }
+
+                _otherExtraFileService.Upsert(result.ToList());
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, $"Failed to import {extraFilePaths.Count()} album extra files for artist '{artist.CleanName}'");
+                return new List<ExtraFile>();
+            }
+        }
+
+        public IEnumerable<ExtraFile> MoveFilesAfterRename(Artist artist, List<TrackFile> trackFiles)
+        {
+            var extraFiles = _otherExtraFileService.GetFilesByArtist(artist.Id);
+            if (!extraFiles.Any())
+            {
+                return new List<ExtraFile>();
+            }
+
+            var movedFiles = new List<OtherExtraFile>();
+
+            try
+            {
+                var albums = trackFiles.GroupBy(x => x.AlbumId);
+                foreach (var albumTracks in albums)
+                {
+                    var albumId = albumTracks.Key;
+                    var newDir = Path.GetDirectoryName(albumTracks.First().Path);
+
+                    var albumExtras = extraFiles.Where(x => x.AlbumId == albumId);
+                    if (!albumExtras.Any())
+                    {
+                        continue;
+                    }
+
+                    foreach (var file in albumExtras)
+                    {
+                        var newPath = MoveExtraFile(artist.Path, file, newDir);
+                        file.RelativePath = artist.Path.GetRelativePath(newPath);
+                        movedFiles.Add(file);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, $"Moving album extras for artist '{artist.CleanName}' failed");
+                return new List<ExtraFile>();
+            }
+
+            _otherExtraFileService.Upsert(movedFiles);
+
+            return movedFiles;
+        }
+
+        private string MoveExtraFile(string artistPath, OtherExtraFile file, string newDir)
+        {
+            var oldPath = Path.Combine(artistPath, file.RelativePath);
+            var fileName = Path.GetFileName(oldPath);
+            var newPath = Path.Combine(newDir, fileName);
+
+            _diskProvider.MoveFile(oldPath, newPath);
+            return newPath;
+        }
+
+        private OtherExtraFile ImportSingleFile(Artist artist, int albumId, string extraFilePath, string targetDir)
+        {
+            var fileName = Path.GetFileName(extraFilePath);
+            var fileExt = Path.GetExtension(extraFilePath);
+
+            var newFileName = Path.Combine(targetDir, fileName);
+            var transferMode = _configService.CopyUsingHardlinks ? TransferMode.HardLinkOrCopy : TransferMode.Copy;
+
+            // TODO: check for readonly / downloaditem etc, adjust transferMode
+            if (extraFilePath != newFileName)
```

Comment:
I would like to avoid this check, but if source and destination are the same, `TransferFile()` will throw an exception.

Reply:
Changed this comparison to work on two DirectoryInfo instances, not perfect but it covers more cases :)